crates/sparrow-compiler/src/types/instruction.rs

```diff
@@ -33,6 +32,16 @@ pub(crate) fn typecheck_inst(
 
             validate_instantiation(&argument_types, signature)
         }
+        InstKind::Udf(udf) => {
+            let signature = udf.signature();
+            let argument_types = Resolved::new(
+                Cow::Owned(signature.parameters().names().to_owned()),
```

Comment:
I don't think I can use `Borrowed` because `udf` is an `Arc`. I thought I would've been able to clone the `Arc`, then use borrowed from that clone's signature, but I was getting `udf not static lifetime`.

Reply:
Actually..I guess that makes sense? `Resolved` requires 
```
        names: Cow<'static, [Located<String>]>,
```
so even if I clone the `Arc`, it is dereferenced at the end of this function, so there's no guarantee the `names()` lives longer than this function call?

Reply:
Yeah -- it would require knowing some lifetime (of the arc) and then using `Cow<'arc, ...>`.

I think if we continue down this road, the better option be for this to just expect `Arc<Signature>` or `Arc<ThingThatHasSignature>` and use the signature from behind the arc.This is where `ArcRef` comes up... but that has issues. 

The other option which may be better is if this doesn't need to return an owned signatuer. If we could instead have:

```
fn signature(&self) -> &Signature { ... }
```

then we can just use the `Arc` that we have to get the signature, and be ok